addon/webextension/background/main.js

```diff
@@ -30,10 +30,14 @@ window.main = (function () {
 
   chrome.browserAction.onClicked.addListener(function(tab) {
     if(tab.url.match(/about:(newtab|blank)/i)) {
-      sendEvent("goto-myshots", "about-newtab");
+      catcher.watchPromise(analytics.refreshTelemetryPref().then(() => {
```

Comment:
Some browser.runtime message is inevitable, since we need to get the pref value to the webextension somehow.